### PR TITLE
[PI-1858] Enhancing celery exporter with auto-discovery

### PR DESCRIPTION
**Pull request by brucehajdu**
## Description
Per [PI-1858](https://abnormalsecurity.atlassian.net/browse/PI-1858), the celery-prometheus exporter does not currently auto-discover new queues.  Adding this ability will be a QoL improvement.  This PR adds that functionality and a couple of ancillary items.

## Changes Made
- Added auto-discovery functionality
- Removed `--queue-list` parameter, as it is not needed with the queue discovery component
- Updated the PyPi package build configuration 
- Updated docker-compose to use Redis instead of RabbitMQ for local testing purposes

## Testing Done
- Queues detect properly and update automatically in local dev environment

## Notes
- Code needs tested properly in dev/staging environment, hence this PR needs merged so we can upload the package, and update the proper PyPi package version in the main celery-exporter Dockerfile

### CODE REVIEW

File: celery_prometheus_exporter.py [hunk 1]
```diff
@@ -214,26 +214,31 @@ def enable_events(self):
...
         super(QueueLengthMonitoringThread, self).__init__()
 
     def measure_queues_length(self):
-        for queue in self.queue_list:
-            try:
-                length = self.connection.default_channel.queue_declare(queue=queue, passive=True).message_count
-            except (amqp.exceptions.ChannelError,) as e:
-                logging.warning("Queue Not Found: {}. Setting its value to zero. Error: {}".format(queue, str(e)))
-                length = 0
+        queue_keys = self.redis_client.scan_iter('*')
```
Comment (kiambogo) [inline celery_prometheus_exporter.py:230]: this is just ignorance and a lack of understanding on my part, but how exactly does listing the keys in redis reveal the list of queues to monitor? is there a redis cluster as part of celery that maintains the queue list?

Comment (brucehajdu) [reply celery_prometheus_exporter.py:231]: From what I saw in the main source code, we're using Redis as the backing store for Celery.  Celery is really just a Python interface to Redis in this case, from what I've seen.  Part of the reason why the exporter was unable to dynamically discover queues before was that it was using the Celery interface, which required prior knowledge of the queue name.  If instead, you instantiate a Redis client, you can do this:

- Get all the keys on the Redis instance
- For each key, perform a `llen` operation on it (which only works if the key's value is of List type)
- If the `llen` operation works, it's a queue, and you can report its length
- If not, it's not a queue, and you can safely ignore it next time (so you're not spamming Redis with bad `llen` operations)

Let me know if that method makes sense, or if I've missed the mark somewhere.

Comment (kiambogo) [reply celery_prometheus_exporter.py:231]: sounds good – thanks!

**Review by kiambogo — APPROVED** [review]
LGTM – lets create some internal documentation on the this work so that others can read up on how the queue auto-discovery works.

**Pull request merged**